src/main/java/io/cdap/plugin/gcp/gcs/sink/GCSBatchSink.java

```diff
@@ -316,7 +330,15 @@ public void validate(FailureCollector collector) {
             .withConfigProperty(NAME_SUFFIX).withStacktrace(e.getStackTrace());
         }
       }
-
+      if (contentType == null) {
+        contentType = DEFAULT_CONTENT_TYPE;
```

Comment:
Shouldn't modify field in the validate method. Instead, introduce "getter" method do the defaulting logic in there. In the validate method, call the getter method for validation if needed.

Reply:
Please check the updated PR.